### PR TITLE
Specs: rework and some speedup

### PR DESCRIPTION
**Pull request by ysbaddaden**
The current spec suite takes a very long time to run (up to 6m50s) and it's discouraging to run it while developping. This pull request doesn't solve the issue, sadly, but improves the situation a little bit (up to 5m20s).

Most of the changes in individual commits didn't improve the situation much. It shave at most 30s from the the total suite run (down to 6m20s). The performance issue seems to be in the factories that instantiate then save records with different callbacks to create the default set of roles & policies, and recomputing the policies, which involves lots of Ruby and SQL requests to run (which are slow).

What significantly sped things up is 03650e0afa420375de9deea4b0ff7546af538da2 where I applied the logic found in `spec/models/blender_spec.rb` to all controllers, to avoid recreating all records before `:each` spec, but instead sharing a set of records created once in before `:all`. It's not very pretty, but it does the trick (down to 5m20s).

Continuing to apply this logic (if we accept it), and using `.make` instead of `.make!` to only instantiate records instead of saving them in model specs should help to further improve the overall run time :crossed_fingers: 

Last but not least: a simple trick to speedup the startup time is to use `bin/rspec` to have spring preload Rails in a background process. The initial overhead goes from ~12s to ~1s, so we can run focused tests repeatedly, without the "damn it takes forever" impact.

### CODE REVIEW

**Comment by ysbaddaden** [comment]
Oh! GHA is faster than my laptop :tada: 
```console
Finished in 4 minutes 36.7 seconds (files took 5.28 seconds to load)
```

File: spec/features/deleted_spec.rb [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
 
 describe "deleted entities should be accessible", elasticsearch: true do
-  let(:institution) { Institution.make }
+  let(:institution) { Institution.make! }
   let(:user) { institution.user }
-  let(:site) { institution.sites.make }
+  let(:site) { institution.sites.make! }
   let!(:device_spec_helper) { DeviceSpecHelper.new 'genoscan' }
-  let!(:device) { device_spec_helper.make site: site }
+  let!(:device) { device_spec_helper.make! site: site }
```
[the remark this answers is not in the document]
Comment (ysbaddaden) [reply spec/features/deleted_spec.rb:9]: They must have slipped me. I have a feeling that those feature specs are never run: the actual integration tests are in `/features` not in `/spec/features`.

File: Gemfile [hunk 2]
```diff
@@ -93,7 +91,8 @@ gem 'dotiw'
 gem 'rails-i18n', '~> 4.0.0'
 gem 'doorkeeper'
 
-gem 'faker'
+gem 'faker' # NOTE: until we upgrade to ruby 2.5+ so we can upgrade to ffaker 2.20 to replace Faker::Number!
```
[the remark this answers is not in the document]
Comment (ysbaddaden) [reply Gemfile:94]: This is what I mean:
1. I'm migrating from the `faker` gem to the `ffaker` gem (slightly faster, zero dependencies);
2. We use Faker::Number in one file, but the equivalent (FFaker::Number) is only available in ffaker 2.20;
3. Issue: ffaker 2.20 requires Ruby 2.5 but we're stuck on Ruby 2.2 right now.

File: spec/features/deleted_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ysbaddaden) [reply spec/features/deleted_spec.rb]: Confirmed. They are never run, not even on CI which only runs the cucumber specs in `/features`. You must type `bin/rspec spec/features` and there are some failures.

File: Gemfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (straight-shoota) [reply Gemfile:94]: Ah, yeah it's a bit twisted. Makes sense now!
Maybe `s/so/then/` to make it more clear?

**Comment by ysbaddaden** [comment]
@straight-shoota Fixed and rebased from `main` that includes the invite institution specs.

File: .rspec [hunk 3]
```diff
@@ -1 +1 @@
---color --format progress --format TestrailRspecFormatter::Formatter --require spec_helper
+--color --format progress --require spec_helper
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply .rspec]: Is `--require spec_helper` really necessary?

**Review by straight-shoota — APPROVED** [review]

**Comment by straight-shoota** [comment]
Sry I didn't realize I had not yet approve this one.

How should we go about merging this? I suppose it might already be out of sync with some recently merged specs?

**Pull request merged**